### PR TITLE
feat(os-14) add rule to check noexec, nosuid and nodev mount options

### PR DESCRIPTION
**Pull request by cmhe**
Setting the `noexec`, `nosuid` and `nodev` mount options for mount
points where those features are not required, limits possible attack
vectors.

Closes: #163

Signed-off-by: Claudius Heine <ch@denx.de>

### CODE REVIEW

File: controls/os_spec.rb [hunk 1]
```diff
@@ -41,6 +41,39 @@
...
 cpuvulndir = '/sys/devices/system/cpu/vulnerabilities/'
 
+mount_exec_blocklist = attribute(
+  'mount_exec_blocklist',
+  value: '/boot:/dev:/dev/shm:/tmp:/var/log:/var/log/audit:/var/tmp:/media:/run/mount',
+  description: 'Colon separated list of paths where \'noexec\' mount option shoud be set'
+)
+mount_exec_approvelist = attribute(
+  'mount_exec_approvelist',
+  value: '/bin:/sbin:/lib:/usr/bin:/usr/sbin:/usr/lib:/usr/libexec:/dev/hugepages',
```
[the remark this answers is not in the document]
Comment (cmhe) [reply controls/os_spec.rb]: THX, changed. I wasn't sure how well inspect handles lists via command line, but it seems to work, albeit more verbose.

File: controls/os_spec.rb [hunk 2]
```diff
@@ -282,3 +315,32 @@
     end
   end
 end
+
+control 'os-14' do
+  impact 1.0
+  title 'Check mount options (noexec, nodev, nosuid)'
+  desc 'Use the noexec, nodev and nosuid mount options to limit attack vectors via mount points'
+
+  inspec.file('/proc/self/mountinfo').content.split(/\n/).map { |l| l.split[4] }.each do |mnt_point|
```
[the remark this answers is not in the document]
Comment (cmhe) [reply controls/os_spec.rb]: I removed the dynamic mount check, and now only check the mount points specified.

I haven't found any precedence of optionally flagged controls in dev-sec, how should we go about that?

[the remark this answers is not in the document]
Comment (chris-rock) [reply controls/os_spec.rb]: You would define an attribute and then add a only_if option if the variable is active

see 
https://github.com/dev-sec/cis-docker-benchmark/blob/master/controls/host_configuration.rb#L52-L70 for inspiration

**Comment by chris-rock** [comment]
This looks great. Thank you @cmhe

**Pull request merged**